cdist/conf/type/__postgres_database/explorer/state

```diff
@@ -20,7 +20,7 @@
 
 name="$__object_id"
 
-if su - postgres -c "echo '\q' | psql '$name'" 2>/dev/null; then
+if su -l postgres -c "echo '\q' | psql '$name'" 2>/dev/null; then
```

Comment:
`psql` accepts commands with `-c`
Additionally, command options should always be before other arguments. Only GNU's `getopt(3)` transparently parses all arguments for options (until `--`). BSDs, among others, stop searching for options at the first non-option argument.

`su -l -c "psql -c '\\q' '$name'" 2>/dev/null` should do the trick

Reply:
For example, FreeBSD su does not have -c option for running command.
-c defines settings of the specified login class.
su synopsis: su [-] [-c class] [-flms] [login [args]]
args are passed to the login shell of the target login.
I took a look at OpenBSD and NetBSD su man pages and they work in that way too.

I think this form will work for GNU and BSDs, at least.
Do you have some other suggestion?